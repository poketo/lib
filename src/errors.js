// @flow

type ErrorCode =
  | 'INVALID_URL'
  | 'UNSUPPORTED_SITE'
  | 'UNSUPPORTED_SITE_REQUEST'
  | 'TIMEOUT'
  | 'HTTP_ERROR';

class PoketoError extends Error {
  code: ErrorCode;
  constructor(errorCode: ErrorCode, message: string) {
    super(message);
    Error.captureStackTrace(this, this.constructor);
    this.name = this.constructor.name;
    this.code = errorCode;
  }
}

class InvalidUrlError extends PoketoError {
  constructor(url: string) {
    super('INVALID_URL', `Unable to parse url '${url}'`);
  }
}

class HTTPError extends PoketoError {
  statusCode: number;
  url: string;

  constructor(statusCode: number, message: string, url: string) {
    super('HTTP_ERROR', message);
    this.statusCode = statusCode;
    this.url = url;
  }
}

class NotFoundError extends HTTPError {
  constructor(url: string) {
    super(404, 'Not Found', url);
  }
}

class TimeoutError extends PoketoError {
  constructor(message: string, url: string) {
    super('TIMEOUT', message);
  }
}

class UnsupportedSiteError extends PoketoError {
  constructor(site: string) {
    super('UNSUPPORTED_SITE', `Site at '${site}' is not supported`);
  }
}

class UnsupportedSiteRequestError extends PoketoError {
  constructor(siteName: string, operationName: string) {
    super(
      'UNSUPPORTED_SITE_REQUEST',
      `${siteName} does not support ${operationName}`,
    );
  }
}

export default {
  PoketoError,
  HTTPError,
  InvalidUrlError,
  NotFoundError,
  TimeoutError,
  UnsupportedSiteError,
  UnsupportedSiteRequestError,
};
