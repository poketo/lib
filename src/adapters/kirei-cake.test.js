import site from './kirei-cake';
import errors from '../errors';

describe('KireiCakeAdapter', () => {
  describe('getSeries', () => {
    it('returns a metadata object', async () => {
      const metadata = await site.getSeries('masamunekuns_revenge');

      expect(metadata).toMatchSnapshot({
        chapters: expect.arrayContaining([
          {
            number: '47',
            createdAt: 1525765063,
            slug: 'en/0/47',
            url:
              'https://reader.kireicake.com/read/masamunekuns_revenge/en/0/47/page/1',
          },
        ]),
      });
    });
  });
});
