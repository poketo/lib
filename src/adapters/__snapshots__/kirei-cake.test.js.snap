// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KireiCake getSeries returns a metadata object 1`] = `
Object {
  "author": "TAKEOKA Hazuki, Tiv",
  "coverImageUrl": undefined,
  "description": "Makabe Masamune has changed himself greatly for a single reason: he wants revenge. Contrary to his past, he has worked hard to become good-looking in order to carry out this goal. Faster than anticipated, he even meets the target of his revenge the first day after transferring in.. Adagaki Aki! He wastes no time beginning his plan, but revenge doesn't come easily.",
  "id": "kirei-cake:masamunekuns_revenge",
  "publicationStatus": "UNKNOWN",
  "site": Object {
    "id": "kirei-cake",
    "name": "Kirei Cake",
  },
  "slug": "masamunekuns_revenge",
  "supportsReading": true,
  "title": "Masamune-kun's Revenge",
  "updatedAt": Any<Number>,
  "url": "http://localhost:57156/series/masamunekuns_revenge",
}
`;

exports[`KireiCake getSeries returns a metadata object 2`] = `
Array [
  Object {
    "chapterNumber": "47",
    "createdAt": 1525765063,
    "id": "kirei-cake:masamunekuns_revenge:en/0/47",
    "slug": "en/0/47",
    "title": undefined,
    "url": "http://localhost:57156/read/masamunekuns_revenge/en/0/47/page/1",
    "volumeNumber": "0",
  },
]
`;
